services/order.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime
+from django.db import transaction
+from db.models import Order, User, MovieSession, Ticket
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict], username: str, date: datetime = None
+) -> Order:
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise ValueError(f"User with username '{username}' does not exist.")
+
+    order = Order.objects.create(

```

Comment:
Instead of checking if date is set and updating the order.created_at after creation, you can set created_at directly in the create() call